Client/Catalog/frontend/store/product/product.repository.ts

```diff
@@ -10,8 +10,9 @@ export class ProductRepository {
 
 	public constructor(private messageBus: MessageBusService) {}
 
-	public findProductsByFilter(): Observable<Product[]> {
-		return this.messageBus.query('FindProductsByFilter', [{}]).pipe(map((response) => this.responseToProducts(response)));
+	public findProductsByFilter(action: any): Observable<Product[]> {
```

Comment:
i think we should only give the arguments we need here, i would give a filter object as parameter but not the whole action